// proto/storage.proto
syntax = "proto3";

package storage;

service StorageService
{
    rpc StoreChunk(StoreChunkRequest) returns(StoreChunkResponse) {}
    rpc RetrieveChunk(RetrieveChunkRequest) returns(RetrieveChunkResponse) {}
    rpc DeleteFile(DeleteFileRequest) returns(DeleteFileResponse) {}
    rpc ListFiles(ListFilesRequest) returns(ListFilesResponse) {}
    rpc HealthCheck(HealthCheckRequest) returns(HealthCheckResponse) {}
}

message StoreChunkRequest
{
    string filename = 1;
    int32 chunk_number = 2;
    bytes data = 3;
    string checksum = 4;
}

message StoreChunkResponse
{
    bool success = 1;
    string message = 2;
}

message RetrieveChunkRequest
{
    string filename = 1;
    int32 chunk_number = 2;
}

message RetrieveChunkResponse
{
    bytes data = 1;
    string checksum = 2;
    bool success = 3;
    string message = 4;
}

message DeleteFileRequest
{
    string filename = 1;
}

message DeleteFileResponse
{
    bool success = 1;
    string message = 2;
}

message ListFilesRequest
{
}

message ListFilesResponse
{
    repeated string filenames = 1;
}

message HealthCheckRequest
{
    string node_id = 1;
}

message HealthCheckResponse
{
    bool healthy = 1;
    double latency_ms = 2;
    string status = 3;
}
